### ISSUE TITLE
Printed SVG image contains newline cutting through XML tags

### ISSUE THREAD

**Issue by ShamrockLee** [root-project/root#9514]
- [X] Checked for duplicates

### Describe the bug

When plotting to SVG file the way
```c++
hist->Draw();
gPad->Print("hist.svg");
```

The produced SVG file is occasionally broken due to the newline character being inserted to the middle of the closing XML tag.

The error might be caused by certain kind of 250-character-per-line limit.

### Expected behavior

The newline should be inserted somewhere before the closing XML tag.

Please don't remove all the new lines just to workaround this bug. It is great to have readable SVG outputs.

### To Reproduce

1.  Prepare the reproduction code
`Reproduce.C`
```c++
#include<TCanvas.h>
#include<TPad.h>
#include<TFile.h>
#include<TH1.h>

void Reproduce(const char *pathFileIn="hsimple.root") {
  TCanvas::MakeDefCanvas();
  TFile *tfIn = TFile::Open(pathFileIn);
  TH1* hpxLongTitle = static_cast<TH1*>(tfIn->Get<TH1>("hpx")->Clone("hpxLongTitle"));
  hpxLongTitle->SetTitle("SupercalifragilisticexpialidociousHonorificabilitudinitatibusFloccinaucinihilipilificationFreeSesquipedalianTitleOnTheSupercalifragilisticexpialidociousHonorificabilitudinitatibusFloccinaucinihilipilificationFreeHist");
  hpxLongTitle->Draw();
  gPad->Print("hpxLongTitle.svg");
  tfIn->Close();
}

int main(int argc, char* argv[]) {
  if ( argc > 1 ) {
    Reproduce(argv[1]);
  } else {
    Reproduce();
  }
}
```

2.  Run by

```console
$ root -l -q 'Reproduce.C++("path/to/hsimple.root")'
```

    or by

```
$ g++ $(root-config --glibs --cflags) -o Reproduce.o Reproduce.C
$ ./Reproduce.o "path/to/hsimple.root"
```

3.  See the generated `hpxLongTitle.svg`
4.  Open with a text editor and scroll to the bottom (pressing `G` for vim)
5.  See the broken XML tag

### Setup

1.  ROOT Version: 6.25/01, built for linuxx8664gcc on Jan 01 1980, 00:00:00
2.  Operating System: NixOS
3.  Build with Nix package manager from master, revision 7821ed1e81a419fda484f01e4d14ce52cb372128

### Additional context

The actual content of that supercali-title in my use case is a JSON string without any space character.

The generated SVG file (add the .txt suffix to workaround GitHub's limitation):
[hpxLongTitle.svg.txt](https://github.com/root-project/root/files/7831387/hpxLongTitle.svg.txt)

**Comment by ShamrockLee** [comment]
In
core/base/src/TVirtualPS.cxx
There goes
```c++
const Int_t kMaxBuffer = 250;
```

**Comment by dpiparo** [comment]
This is not an area where I would call myself an expert, by far. However, I have a naive observation: wouldn't the fix just consist in raising the limit of kMaxBuffer to something far higher than 250 and maybe add a check to that number to stop the svg production if the string is longer than that limit?

**Comment by couet** [comment]
Fix here: https://github.com/root-project/root/issues/9514

**Comment by ShamrockLee** [comment]
> Fix here: #9514

Cool!

**Comment by couet** [comment]
PR Merged
